lib/vector-core/src/event/metric/value.rs

```diff
@@ -480,23 +513,32 @@ impl From<MetricValue> for ::value::Value {
     }
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Deserialize, Serialize)]
+/// Type of statistics to generate for a distribution.
+#[configurable_component]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd)]
 #[serde(rename_all = "snake_case")]
 pub enum StatisticKind {
+    /// A histogram representation.
     Histogram,
+
     /// Corresponds to Datadog's Distribution Metric
     /// <https://docs.datadoghq.com/developers/metrics/types/?tab=distribution#definition>
     Summary,
 }
 
-#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+/// A generalized metrics sketch.
+#[configurable_component]
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub enum MetricSketch {
-    /// DDSketch implementation based on the Datadog Agent.
+    /// [DDSketch][ddsketch] implementation based on the [Datadog Agent][ddagent].
     ///
     /// While DDSketch has open-source implementations based on the white paper, the version used in
-    /// the Datadog Agent itself is subtly different.  This version is suitable for sending directly
+    /// the Datadog Agent itself is subtly different. This version is suitable for sending directly
     /// to Datadog's sketch ingest endpoint.
-    AgentDDSketch(AgentDDSketch),
+    ///
+    /// [ddsketch]: https://www.vldb.org/pvldb/vol12/p2195-masson.pdf
+    /// [ddagent]: https://github.com/DataDog/datadog-agent
```

Comment:
We're doing this style links, not due to a requirement, but to avoid escaping characters in the `[]()` style?

Reply:
Primarily to avoid having huge, long links inline with the docs, because it makes the line breaks look fucky and ends up being much harder to read, IMO. It also lets us reuse a single link reference definition within the same doc comment block.

Reply:
Realized while looking at this, though, that I didn't even use the links in another spot where I copied these links (specifically the docs for `AgentDDSketch` itself) so I fixed that in 8b55d4493.